### PR TITLE
Add IPv6 support for P2P integration tests

### LINKED ISSUE #1833: Write a test for IPv6
1) establishing connection + basic communication
2) addrbook handles IPv6 IPs properly
3) pex handles IPv6 IPs properly

### PR DESCRIPTION
**Pull request by erikgrinaker**
Modifies the existing P2P integration test suite to support running on an IPv6 network stack. Also removes `cd`-calls that assume use of `GOPATH` instead of Go modules.

IPv6 tests require IPv6 to be enabled in the Docker daemon. To allow developers to easily run local tests without fiddling with Docker settings, the `make test_integrations` target does not run IPv6 tests by default - they must be executed manually via `make test_p2p_ipv6`.

CircleCI will run IPv6 tests by default.

Fixes #1833.

* [x] Referenced an issue explaining the need for the change
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] ~Updated CHANGELOG_PENDING.md~

### CODE REVIEW

File: test/p2p/basic/test.sh [hunk 1]
```diff
@@ -16,7 +17,8 @@ MAX_SLEEP=60
 # wait for everyone to come online
 echo "Waiting for nodes to come online"
 for i in `seq 1 $N`; do
-	addr=$(test/p2p/ip.sh $i):26657
+	addr=$(test/p2p/address.sh $IPV $i 26657)
+	echo $addr
```
Comment (melekes) [inline test/p2p/basic/test.sh]: leftover?

File: tests.mk [hunk 2]
```diff
@@ -49,6 +49,27 @@ test_p2p:
...
+	# {
+	#	"ipv6": true,
+	#   "fixed-cidr-v6": "2001:db8:1::/64"
+	# }
+	#
+	# Docker for Mac can set this via Preferences -> Docker Engine.
+	docker rm -f rsyslog || true
+	rm -rf test/logs || true
+	mkdir test/logs
+	cd test/
```
Comment (melekes) [inline tests.mk]: won't it be better to chain the following three commands?

```
cd test/ &&
  docker ... &&
  cd ..
```

File: test/p2p/basic/test.sh (hunk 1 above)
Comment (erikgrinaker) [reply test/p2p/basic/test.sh]: Yes, good catch.

File: tests.mk (hunk 2 above)
Comment (erikgrinaker) [reply tests.mk]: I removed the cd'ing altogether. This was just copied from `test_p2p`.

File: tests.mk [hunk 3]
```diff
@@ -38,11 +38,8 @@ test_persistence:
 
 test_p2p:
 	docker rm -f rsyslog || true
-	rm -rf test/logs || true
-	mkdir test/logs
-	cd test/
-	docker run -d -v "logs:/var/log/" -p 127.0.0.1:5514:514/udp --name rsyslog voxxit/rsyslog
-	cd ..
+	rm -rf test/logs && mkdir test/logs
```
Comment (melekes) [inline tests.mk]:
```suggestion
	rm -rf test/logs && mkdir -p test/logs
```

File: tests.mk [hunk 4]
```diff
@@ -59,11 +56,8 @@ test_p2p_ipv6:
 	#
 	# Docker for Mac can set this via Preferences -> Docker Engine.
 	docker rm -f rsyslog || true
-	rm -rf test/logs || true
-	mkdir test/logs
-	cd test/
-	docker run -d -v "logs:/var/log/" -p 127.0.0.1:5514:514/udp --name rsyslog voxxit/rsyslog
-	cd ..
+	rm -rf test/logs && mkdir test/logs
```
Comment (melekes) [inline tests.mk]:
```suggestion
	rm -rf test/logs && mkdir -p test/logs
```

**Pull request merged**